osu.Game/Screens/Select/FooterV2/FooterButtonV2.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osu.Framework.Graphics.Sprites;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Input.Events;
+using osu.Framework.Localisation;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Input.Bindings;
+using osu.Game.Overlays;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Screens.Select.FooterV2
+{
+    public partial class FooterButtonV2 : OsuClickableContainer, IKeyBindingHandler<GlobalAction>
+    {
+        private const int button_height = 90;
+        private const int button_width = 140;
+        private const int corner_radius = 10;
+        private const int transition_length = 500;
+
+        // This should be 12 by design, but an extra allowance is added due to the corner radius specification.
+        public const float SHEAR_WIDTH = 13.5f;
+
+        public Bindable<Visibility> OverlayState = new Bindable<Visibility>();
+
+        protected static readonly Vector2 SHEAR = new Vector2(SHEAR_WIDTH / button_height, 0);
+
+        [Cached]
+        private OverlayColourProvider colourProvider = new OverlayColourProvider(OverlayColourScheme.Aquamarine);
+
+        private Colour4 buttonAccentColour;
+
+        protected Colour4 AccentColour
+        {
+            set
+            {
+                buttonAccentColour = value;
+                bar.Colour = buttonAccentColour;
+                icon.Colour = buttonAccentColour;
+            }
+        }
+
+        protected IconUsage Icon
+        {
+            set => icon.Icon = value;
+        }
+
+        protected LocalisableString Text
+        {
+            set => text.Text = value;
+        }
+
+        private readonly SpriteText text;
+        private readonly SpriteIcon icon;
+
+        protected Container TextContainer;
+        private readonly Box bar;
+        private readonly Box backgroundBox;
+
+        public FooterButtonV2()
+        {
+            EdgeEffect = new EdgeEffectParameters
+            {
+                Type = EdgeEffectType.Shadow,
+                Radius = 5,
+                Roundness = 10,
```

Comment:
Not sure if roundness is needed, see xmldoc:
> Adds to the CornerRadius of the corresponding CompositeDrawable.
If it is:
```suggestion
                Roundness = corner_radius,
```